### ISSUE TITLE
Flutter API docs included libraries missing link to code

### ISSUE THREAD

**Issue by guidezpl** [dart-lang/dartdoc#3756]
I noticed some libraries included in Flutter don't have a `View source code` button e.g. https://api.flutter.dev/flutter/dart-ui/Color/lerp.html

Is this a limitation or can be worked around?

<img width="2081" alt="Screenshot 2024-04-23 at 09 35 34" src="https://github.com/dart-lang/dartdoc/assets/6655696/8a7d7b8d-e710-4ed5-b6ad-09eec764b99f">

**Comment by srawlins** [comment]
Yes, it looks like no methods, etc. from `dart:ui` have a "Show source code" in the top right of the main frame. But methods found in package:flutter _do_ have those links. @goderbauer do you know if this is expected / intentional? Should only package:flutter have such links? Should all packages? Should a short list of packages have those links?

**Comment by goderbauer** [comment]
Ideally, all packages would have source links. We currently seem to only set up source links for the flutter/flutter repository, though, see: https://github.com/flutter/flutter/blob/b0524b354b1593e17f292cb4e78145916a549898/dev/tools/create_api_docs.dart#L570-L575

And after dartdoc is done we do some post-processing on the links here to "fix" them: https://github.com/flutter/flutter/blob/b0524b354b1593e17f292cb4e78145916a549898/dev/tools/create_api_docs.dart#L298

Not sure what we'd have to do to link to source for other packages (including the engine). Does dart doc support that? The docs are a little vague about this: https://github.com/dart-lang/dartdoc/blob/d9e31ff950c92350b464b859eaac272a97e15070/README.md?plain=1#L447-L451
